test/cri-containerd/container_test.go

```diff
@@ -315,6 +317,228 @@ func Test_RunContainer_VirtualDevice_GPU_LCOW(t *testing.T) {
 	}
 }
 
+func Test_RunContainer_VirtualDevice_GPU_Multiple_LCOW(t *testing.T) {
```

Comment:
I think we are missing the test to check if the ref counting works correctly when containers are removed. We should add another test where we create multiple containers with the GPU and then remove those containers one by one and make sure that one of the existing container still sees the GPU device and then after that last container is removed, create another container and then verify that that container does not see the device.

Reply:
I like it :) I'll add that.

Reply:
Updated. I've moved the GPU tests to their own file and refactored out some code duplication.